src/Common/src/Interop/OSX/Interop.CoreFoundation.CFData.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+// Declared as signed long, which has sizeof(void*) on OSX.
+using CFIndex=System.IntPtr;
+
+internal static partial class Interop
+{
+    internal static partial class CoreFoundation
+    {
+        [DllImport(Libraries.CoreFoundationLibrary)]
+        private static extern unsafe byte* CFDataGetBytePtr(SafeCFDataHandle cfData);
+
+        [DllImport(Libraries.CoreFoundationLibrary)]
+        private static extern CFIndex CFDataGetLength(SafeCFDataHandle cfData);
+
+        internal static byte[] CFGetData(SafeCFDataHandle cfData)
```

Comment:
> was either of them what you were thinking?

I had in mind the latter, or more specifically a func variation of it, e.g.
```C#
internal static TResult CFReadData<TResult>(SafeCFDataHandle cfData, Func<byte*,int,TResult> reader);
```
though it looks like maybe there are just two realistic forms of this, one returning a byte[] and one returning a string? In which case we could just have CFGetData that returns a byte[] and CFGetString that returns a string (and that's not built in terms of the former).